### ISSUE TITLE
UnresolvedAddressException in AsyncHttp

### ISSUE THREAD

**Issue by roundhill** [Simperium/simperium-android#156]
WordPress client has seen a few crashes:

```
java.nio.channels.UnresolvedAddressException
       at java.nio.SocketChannelImpl.validateAddress(SocketChannelImpl.java:431)
       at java.nio.SocketChannelImpl.connect(SocketChannelImpl.java:163)
       at com.koushikdutta.async.AsyncServer$6.run(AsyncServer.java:358)
       at com.koushikdutta.async.AsyncServer.lockAndRunQueue(AsyncServer.java:683)
       at com.koushikdutta.async.AsyncServer.runLoop(AsyncServer.java:700)
       at com.koushikdutta.async.AsyncServer.run(AsyncServer.java:608)
       at com.koushikdutta.async.AsyncServer.access$700(AsyncServer.java:37)
       at com.koushikdutta.async.AsyncServer$13.run(AsyncServer.java:557)
```

```
thread
       at java.lang.Object.wait(Object.java)
       at java.lang.Object.wait(Object.java:364)
       at com.simperium.client.Channel$ChangeProcessor.run(Channel.java:1302)
       at java.lang.Thread.run(Thread.java:856)
```

**Comment by beaucollins** [comment]
Any common device or SDK version?

Stack trace confuses me a bit.

**Comment by beaucollins** [comment]
This may require a fix in the android async library 

https://github.com/koush/AndroidAsync/blob/master/AndroidAsync/src/com/koushikdutta/async/AsyncServer.java#L358

**Comment by maxme** [comment]
No common device. Crash happens with wpandroid (Simperium 0.5.0) but not with wpandroid 3.0 (Simperium 0.4.8)

**Comment by beaucollins** [comment]
@maxme can you release a build with Simperium `0.4.12` (and perhaps lock the dependency to `0.4.+`).

Its use of `ConnectivityManager` should sidestep this issue.

**Comment by maxme** [comment]
Ok. 

I can't find the 0.5.0 tag in the repo, but I tried to upgrade a wpandroid using simperium 0.5.0 to a wpandroid using simperium 0.4.12 and everything went fine. Can you confirm there is no DB downgrade issue?

**Comment by beaucollins** [comment]
@maxime there's no DB issue. 0.5.0 should be 0.5.0-alpha. The artifact in maven is a naming error on my part, sorry about that.

**Comment by beaucollins** [comment]
Filed koush/AndroidAsync#237 to see if AndroidAsync should handle these `RuntimeException`s.

**Comment by maxme** [comment]
FYI: same problem using version 0.4.12

**Comment by beaucollins** [comment]
@maxme  It has been fixed in AndroidAsync. I'll let you know when we have a new build.

**Comment by beaucollins** [comment]
New AndroidAsync has been available with fix since version 1.4.0
